tools/cellxgene_census_builder/src/cellxgene_census_builder/__main__.py

```diff
@@ -150,11 +155,30 @@ def do_prebuild_checks(args: CensusBuildArgs) -> bool:
     return True
 
 
+def do_log_environment(args: CensusBuildArgs) -> bool:
+    # Log pip freeze output
+    import pkg_resources
+
+    deps = ""
+    for dep in pkg_resources.working_set:
+        deps += f"{dep.as_requirement()}\n"
+    logging.info(f"Pip freeze:\n{deps}")
+
+    # Log TileDB Config
+    logging.info(f"TileDB Config:\n{DEFAULT_TILEDB_CONFIG}")
+
+    return True
+
+
 def do_build_soma(args: CensusBuildArgs) -> bool:
     from .build_soma import build as build_a_soma
 
     if (cc := build_a_soma(args)) != 0:
         logging.critical(f"Build of census failed with code {cc}.")
+
```

Comment:
What is the rationale for putting this logging in the package, vs. in the docker `entrypoint.sh`?  We currently do `dmesg` loggging in entrypoint.sh, and it would be good to be consistent.

My _opinion_ is that it belongs in entrypoint.sh, at the end of the run (as in the `dmesg` logging).  Rationale: we only need this introspection when running in Batch/container. Any other time, the user is running from the CLI and can do this type of system introspection on their own.

Reply:
https://github.com/chanzuckerberg/cellxgene-census/blob/deceee6411286d0f8915d5100c39a2d4621f0301/tools/cellxgene_census_builder/entrypoint.sh#L7

Reply:
I don't like putting it into the mainline script, as it isn't needed except for debugging the container.

That said, I'm also not too opinionated.  @atolopko-czi established the convention of putting container/EBS specific stuff into entrypoint.sh, so I'll let you guys hash it out.